api/src/main/java/org/openmrs/CohortMembership.java

```diff
@@ -0,0 +1,100 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import java.util.Date;
+
+public class CohortMembership extends BaseOpenmrsData implements Comparable<CohortMembership> {
```

Comment:
I hope semantics you mention is for the contains method. If I'm not wrong, the `contains` method should look for the patientId and should be an active member, considering there are no duplicate patientIds. So how will the `contains` look like? Considering the active member constitutes the `startDate` to be before the current DATETIME and the `endDate` is `null`?

```
public boolean contains(Integer patientId) {
		return getMembers() != null && getMembers().stream()
				.anyMatch(m -> m.getPatient().getPatientId().equals(patientId) &&
				m.getStartDate().before(new Date()) &&
				m.getEndDate() == null);
	}
```